src/sections/create-dataset/CreateDatasetContext.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react'
+import {
+  CreateDatasetFormFields,
+  FormSubmissionService,
+  FormValidationService,
+  FormValidationResult
+} from '../../dataset/domain/useCases/createDataset'
+
+interface FormContextInterface {
+  fields: CreateDatasetFormFields
+}
+
+const defaultFormState: CreateDatasetFormFields = {
+  createDatasetTitle: ''
+}
+
+interface CreateDatasetFormProps {
+  formValidationService: FormValidationService
+  formSubmissionService: FormSubmissionService
+}
+
+export enum SubmissionStatusEnums {
+  NotSubmitted = 'NotSubmitted',
+  IsSubmitting = 'IsSubmitting',
+  SubmitComplete = 'SubmitComplete',
+  Errored = 'Errored'
+}
+
+export function useCreateDatasetForm({
+  formValidationService,
+  formSubmissionService
+}: CreateDatasetFormProps) {
+  const [formState, setFormState] = useState<FormContextInterface>({
+    fields: defaultFormState
+  })
+  const [submissionStatus, setSubmissionStatus] = React.useState<SubmissionStatusEnums>(
+    SubmissionStatusEnums.NotSubmitted
+  )
+  const [formErrors, setFormErrors] = useState<
+    Record<keyof CreateDatasetFormFields, string | undefined>
+  >({ createDatasetTitle: undefined })
+
+  const handleCreateDatasetFieldChange = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
I apologize if I wasn't clear in our last review. The `useCreateDatasetForm` hook is responsible for the data handling, whereas the `CreateDataset` component is responsible for reading the data from the hook and rendering the UI . Given that `HTMLInputElement` pertains to the UI, it is necessary to maintain the HTML listener within the `CreateDataset` component file. Meanwhile, the `useCreateDatasetForm` hook will provide an `updateFormData` function, which is not coupled to HTML elements, for updating the form data

So we move this part to the `CreateDataset` component

```
const { updateFormData } = useCreateDatasetForm()
const handleCreateDatasetFieldChange = (event: ChangeEvent<HTMLInputElement>) => {
const { name, value } = event.target

updateFormData({ [name]: value });
}
```

And in this hook we expose a function to update the formData state:
```
const updateFormData = (updatedFormData) => {
setFormState((prevState) => ({
       ...prevState,
       ...updatedFormData
     }))
   }
```

Reply:
I think that Ellen and I fixed these issues together late last week and the new changes should be ready to look over. If you have any feedback lets throw a meeting together and get it worked out :)